File: chat-message-server/src/main/java/com/example/chat/message/infrastructure/messaging/MessageSentEvent.java
package com.example.chat.message.infrastructure.messaging;

import lombok.Builder;
import lombok.Getter;

import java.time.Instant;

/**
 * 메시지 발송 이벤트 DTO
 * Redis Pub/Sub으로 전송되는 이벤트
 */
@Getter
@Builder
public class MessageSentEvent {

    private String messageId;     // String (UUID)
    private String channelId;     // ChannelId
    private String senderId;      // UserId
    private String messageType;   // MessageType name
    private String content;       // 텍스트 내용
    private String status;        // MessageStatus name
    private Instant sentAt;
}

File: apps/chat/message-server/src/main/kotlin/com/example/chat/message/infrastructure/messaging/MessageSentEvent.kt
package com.example.chat.message.infrastructure.messaging

import java.time.Instant

/**
 * 메시지 발송 이벤트 DTO
 * Redis Pub/Sub으로 전송되는 이벤트
 *
 * 불변 객체로 설계 (Value Object)
 */
data class MessageSentEvent(
	val messageId: String,     // String (UUID)
	val channelId: String,     // ChannelId
	val senderId: String,      // UserId
	val messageType: String,   // MessageType name
	val content: String,       // 텍스트 내용
	val status: String,        // MessageStatus name
	val sentAt: Instant?
)
